From ecc2b5b77911681de5e445e49bf71ee421b799df Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio.traversaro@iit.it>
Date: Sun, 10 Apr 2022 22:49:29 +0200
Subject: [PATCH] Generator: Add missing std namespace to string arguments

---
 src/Generator.hh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/Generator.hh b/src/Generator.hh
index c0f2336f..61322741 100644
--- a/src/Generator.hh
+++ b/src/Generator.hh
@@ -44,9 +44,9 @@ class Generator : public CodeGenerator
   /// \param[in] _generatorContext Output directory.
   /// \param[in] _error Unused string value
   public: virtual bool Generate(const FileDescriptor *_file,
-              const string &_parameter,
+              const std::string &_parameter,
               OutputDirectory *_generatorContext,
-              string *_error) const;
+              std::string *_error) const;
 
   // private: GOOGLE_DISALLOW_EVIL_CONSTRUCTORS(Generator);
 };
